### ISSUE TITLE
Unsanitized "$lt;" still occurs

### ISSUE THREAD

**Issue by mckbrchill** [huggingface/chat-ui#533]
I mentioned that "$lt;" still occurs in code blocks

![image](https://github.com/huggingface/chat-ui/assets/115869584/ebbaa84b-4268-4ffd-abc6-cb3ce96f67ff)

It seems it happens when the code block is in list token, parsed by marked.
So it falls into else statement here, because token.type == list, and it contains code inside:

```
{#if token.type === "code"}
	<CodeBlock lang={token.lang} code={unsanitizeMd(token.text)} />
{:else}
	<!-- eslint-disable-next-line svelte/no-at-html-tags -->
	{@html marked(token.raw, options)}
{/if}
```

**Comment by mckbrchill** [comment]
Yep, it's because the code block is nested inside list item, moreover this item list also has subtokens (item.tokens) and code token is one of them, so the unsanitizeMd doesn't touch it.
Also because of that the code is rendered as a default markdown without code language highlight.

**Comment by nsarrazin** [comment]
Thanks for reporting this, I guess indeed we need some kind of recursive traversing of tokens, not sure how that would work 🤔
